File: C++/QuickSort.cpp
#include <iostream>
using namespace std;

int partition(int a[],int strt,int end){
	int count=0;
	int pivot = a[strt];
	for(int i=strt;i<=end;i++){
		if(a[i]<pivot)
			count++;
	}
	int temp = a[count+strt];
	a[count+strt]=a[strt];
	a[strt] = temp;
	
	int pi = count+strt;
	
	int i=strt,j=end;
	while (i<pi and j>pi){
		if(a[i]<=a[pi]){
			i++;
		}
		else if(a[j]>= a[pi]){
			j--;
		}
		else{
			int temp = a[j];
			a[j]=a[i];
			a[i]=temp;
		}
	}
	return pi;
}

void QuickSort(int a[],int strt,int end){
	if(strt < end ){
		int c=partition(a,strt,end);
		QuickSort(a,strt,c-1);
		QuickSort(a,c+1,end);
	}
}

int main(){
	int n;
	cout<<"Enter the size of an array :";
	cin>>n;
	int *arr = new int [n];
	cout<<"Enter the array elements :";
	for(int i=0;i<n;i++){
		cin>>arr[i];
	}
	cout<<"Array before sorting :";
	for(int i=0;i<n;i++){
		cout<<arr[i]<<" ";
	}
	cout<<endl;
	QuickSort(arr,0,n-1);
	cout<<"Array after sorting :";
	for(int i=0;i<n;i++){
		cout<<arr[i]<<" ";
	}
	cout<<endl;
	delete []arr;
	return 0;
}

File: Python/quicksort.py
def partition(start, end, array):
      
    pivot_index = start 
    pivot = array[pivot_index]
      
    while start < end:
          
        while start < len(array) and array[start] <= pivot:
            start += 1
              
        while array[end] > pivot:
            end -= 1
          
        if(start < end):
            array[start], array[end] = array[end], array[start]
      
    array[end], array[pivot_index] = array[pivot_index], array[end]
     
    return end
      
def quick_sort(start, end, array):
      
    if (start < end):
        p = partition(start, end, array)
        quick_sort(start, p - 1, array)
        quick_sort(p + 1, end, array)

arr = []
n = int(input("Number of elements in the array: "))
print("Enter each elemnt on one line: ");
for i in range(n):
    temp = input("> ")
    if temp.isnumeric():
        temp = int(temp)
    arr.append(temp)
quick_sort(0,n-1,arr);
print(f'Sorted array: {arr}')